packages/twenty-server/src/engine/core-modules/billing/webhooks/services/billing-webhook-alert.service.ts

```diff
@@ -0,0 +1,68 @@
+/* @license Enterprise */
+
+import { Injectable, Logger } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import Stripe from 'stripe';
+import { isDefined } from 'twenty-shared/utils';
+import { Repository } from 'typeorm';
+
+import {
+  BillingException,
+  BillingExceptionCode,
+} from 'src/engine/core-modules/billing/billing.exception';
+import { BillingProduct } from 'src/engine/core-modules/billing/entities/billing-product.entity';
+import { BillingSubscriptionItem } from 'src/engine/core-modules/billing/entities/billing-subscription-item.entity';
+import { BillingSubscription } from 'src/engine/core-modules/billing/entities/billing-subscription.entity';
+import { SubscriptionStatus } from 'src/engine/core-modules/billing/enums/billing-subscription-status.enum';
+
+const TRIAL_PERIOD_ALERT_TITLE = 'TRIAL_PERIOD_ALERT'; // title used in Stripe config
+
+@Injectable()
+export class BillingWebhookAlertService {
+  protected readonly logger = new Logger(BillingWebhookAlertService.name);
+  constructor(
+    @InjectRepository(BillingSubscription, 'core')
+    private readonly billingSubscriptionRepository: Repository<BillingSubscription>,
+    @InjectRepository(BillingProduct, 'core')
+    private readonly billingProductRepository: Repository<BillingProduct>,
+    @InjectRepository(BillingSubscriptionItem, 'core')
+    private readonly billingSubscriptionItemRepository: Repository<BillingSubscriptionItem>,
+  ) {}
+
+  async processStripeEvent(data: Stripe.BillingAlertTriggeredEvent.Data) {
+    const { customer: stripeCustomerId, alert } = data.object;
+
+    const stripeMeterId = alert.usage_threshold?.meter as string | undefined;
+
+    if (alert.title === TRIAL_PERIOD_ALERT_TITLE && isDefined(stripeMeterId)) {
```

Comment:
i'm unclear about alert stripe capabilities and what will be in stripe and what will be on our side